tutorials/cloud-functions-firestore/index.md

```diff
@@ -0,0 +1,256 @@
+
+You can create a new document:
+
+    curl --header "Content-Type: application/json" \
+      --request POST \
+      --data '{"ttl":1,"ciphertext":"daa5370871aa301e5e12d4274d80691f75e295d648aa84b73e291d8c82"}' \
+      http://localhost:8010/cloud-functions-firestore/us-central1/main
+    {"id":"wLcIOzic6BeoEk3tV4sH"}
+
+And retrieve that document:
+
+    curl http://localhost:8010/cloud-functions-firestore/us-central1/main?id=wLcIOzic6BeoEk3tV4sH
+    {"created":15369690190000,"ttl":1,"ciphertext":"daa5370871aa301e5e12d4274d80691f75e295d648aa84b73e291d8c82"}
+
+NOTE that our function code added `created`.
+
+### Deploy the Function to GCP Functions
+
+This is very easy thanks to the gcloud CLI.
+
+    gcloud functions deploy main --trigger-http
+
+In this case, `main` is the name of the function you want to trigger in your code, triggered by a HTTP request.
+
+> NOTE: If you used the emulator above, you can simply prefix the command with `gcloud`
+
+    Deploying function (may take a while - up to 2 minutes)...done.
+    availableMemoryMb: 256
+    entryPoint: main
+    httpsTrigger:
+      url: https://us-central1-cloud-functions-firestore.cloudfunctions.net/main
+    labels:
+      deployment-tool: cli-gcloud
+    name: projects/cloud-functions-firestore/locations/us-central1/functions/main
+    runtime: nodejs6
+    serviceAccountEmail: cloud-functions-firestore@appspot.gserviceaccount.com
+    sourceUploadUrl: https://storage.googleapis.com/gcf-upload-us-central1-0000000000.zip?GoogleAccessId=service-...
+    status: ACTIVE
+    timeout: 60s
+    updateTime: '2018-09-15T01:25:59Z'
+    versionId: '1'
+
+## Testing the Deployed Function in Production
+
+You should now be able to send HTTP requests to the endpoint and test out the function in production.
+
+Now you can test your function by sending `curl` requests.
+
+You can create a new document:
+
+    curl --header "Content-Type: application/json" \
+      --request POST \
+      --data '{"ttl":1,"ciphertext":"daa5370871aa301e5e12d4274d80691f75e295d648aa84b73e291d8c82"}' \
+      https://us-central1-cloud-functions-firestore.cloudfunctions.net/main
+    {"id":"wLcIOzic6BeoEk3tV4sH"}
+
+And retrieve that document:
+
+    curl https://us-central1-cloud-functions-firestore.cloudfunctions.net/main?id=wLcIOzic6BeoEk3tV4sH
+    {"created":15369690190000,"ttl":1,"ciphertext":"daa5370871aa301e5e12d4274d80691f75e295d648aa84b73e291d8c82"}
+
+NOTE that our function code added `created`.
+
+## Iterate
+
+This is a very short cycle of `[code, deploy, test]`
+so you should be able to iterate rapidly.
+
+When you deploy, you overwrite the current version, at the function's URL *(blue/green deployment in the background, can take a few seconds to switch over)*.
+
+If you need to access a previously deployed version, you can append `/revisions/${REVISION}` with the value of the `versionId` that deploy gives you back.
+
+ There are lot of other settings available, review docs and help
```

Comment:
This is two sentences. Replace the comma with a period. End the second sentence with a colon, to lead into the command that follows.

Reply:
done